lib/services/elasticsearch.js

```diff
@@ -389,22 +381,27 @@ function ElasticSearch (kuzzle, options, config) {
    * @param {Request} request
    * @returns {Promise} resolve an object that contains _id
    */
-  this.update = function elasticsearchUpdate (request) {
-    var esRequest = getElasticsearchRequest(request, kuzzle, ['refresh']);
+  update(request) {
+    const esRequest = getElasticsearchRequest(request, this.kuzzle, ['refresh']);
 
     if (esRequest.body._routing) {
       return Promise.reject(new BadRequestError('Kuzzle does not support "_routing" in update action.'));
     }
 
     if (esRequest.hasOwnProperty('refresh')) {
-      if (compareVersions(config.apiVersion, '5.0') < 0) {
-        return Promise.reject(new BadRequestError(`Refresh parameter is not supported by the version "${config.apiVersion}" of Elasticsearch API`));
+      if (compareVersions(this.config.apiVersion, '5.0') < 0) {
+        return Promise.reject(new BadRequestError(`Refresh parameter is not supported by the version "${this.config.apiVersion}" of Elasticsearch API`));
       }
       if (esRequest.refresh !== 'wait_for' && esRequest.refresh !== 'false' && esRequest.refresh !== false) {
         return Promise.reject(new BadRequestError('Refresh parameter only supports the value "wait_for" or false'));
       }
     }
 
+    // retryOnConflict configuration
+    if (this.config.defaults.onUpdateConflictRetries > 0) {
+      esRequest.retryOnConflict = this.config.defaults.onUpdateConflictRetries;
+    }
```

Comment:
Can we have a way to specify this setting through an API parameter (like `refresh`)